api/http/handler/stacks/create_compose_stack.go

```diff
@@ -146,17 +147,19 @@ func (handler *Handler) createComposeStackFromGitRepository(w http.ResponseWrite
 
 	gitCloneParams := &cloneRepositoryParameters{
 		url:            payload.RepositoryURL,
+		referenceName:  payload.RepositoryReferenceName,
 		path:           projectPath,
 		authentication: payload.RepositoryAuthentication,
 		username:       payload.RepositoryUsername,
 		password:       payload.RepositoryPassword,
 	}
 	err = handler.cloneGitRepository(gitCloneParams)
+	doCleanUp := true
```

Comment:
Good idea to trigger the clean-up right after the clone in case of problem, I believe it will solve the problem raised in https://github.com/portainer/portainer/issues/2050#issuecomment-406806141